File: backend/car-catalog-backend/src/main/java/com/wswork/carcatalog/dto/CarroRespostaDTO.java
package com.wswork.carcatalog.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * DTO para resposta de listagem de carros
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CarroRespostaDTO {
    
    private Long id;
    private String nomeModelo;
    private String nomeMarca;
    private Integer ano;
    private String combustivel;
    private Integer numPortas;
    private String cor;
    private BigDecimal valor;
    private Long timestampCadastro;
    private LocalDateTime dataCriacao;
    private LocalDateTime dataAtualizacao;
}

File: backend-kotlin/car-catalog-kotlin/src/main/kotlin/com/wswork/carcatalog/dto/CarroRespostaDTO.kt
package com.wswork.carcatalog.kotlin.dto

data class CarroRespostaDTO(
    val id: Long,
    val nomeModelo: String,
    val nomeMarca: String,
    val ano: Int,
    val combustivel: String,
    val numPortas: Int,
    val cor: String,
    val valor: Double,
    val timestampCadastro: Long,
    val dataCriacao: String,
    val dataAtualizacao: String
)
